File: frontend/models/Kategori.php
<?php
require_once __DIR__ . '/../plugins/api.php';

class Kategori
{
    public static function getAll()
    {
        $response = api_request('GET', 'kategori');

        if (isset($response['success']) && $response['success']) {
            return $response['data'] ?? [];
        }
        return [];
    }

    public static function getById($id)
    {
        $response = api_request('GET', 'kategori/' . $id);

        if (isset($response['success']) && $response['success']) {
            return $response['data'];
        }
        return null;
    }

    public static function create($ket_kategori)
    {
        $payload = [
            'ket_kategori' => $ket_kategori
        ];

        $response = api_request('POST', 'kategori', $payload);

        if (isset($response['success']) && $response['success']) {
            return $response['data'];
        }
        return false;
    }
}
?>

File: backend/handlers/kategori.go
package handlers

import (
	"encoding/json"
	"log"
	"net/http"

	"pengaduan/database"
	"pengaduan/models"
)

// GetAllKategori retrieves all categories
func GetAllKategori(w http.ResponseWriter, r *http.Request) {
	var kategoriList []models.Kategori
	data, _, err := database.DB.From("kategori").
		Select("id_kategori, ket_kategori", "exact", false).
		Execute()
	if err != nil {
		http.Error(w, "Gagal parsing data", http.StatusInternalServerError)
		return
	}

	if err := json.Unmarshal(data, &kategoriList); err != nil {
		http.Error(w, "Gagal parsing data", http.StatusInternalServerError)
		return
	}

	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(models.Response{
		Success: true,
		Message: "Categories retrieved",
		Data:    kategoriList,
	})
}

// CreateKategori creates a new category
func CreateKategori(w http.ResponseWriter, r *http.Request) {
	var kategori models.Kategori
	err := json.NewDecoder(r.Body).Decode(&kategori)
	if err != nil {
		log.Printf("Error Decode JSON: %v", err)
		http.Error(w, "Invalid request payload", http.StatusBadRequest)
		return
	}

	var result []models.Kategori
	data, _, err := database.DB.From("kategori").
		Insert(map[string]interface{}{
			"ket_kategori": kategori.KetKategori,
		}, false, "", "", "").
		Execute()
	if err != nil {
		http.Error(w, "Failed to create category", http.StatusInternalServerError)
		return
	}

	if err := json.Unmarshal(data, &result); err != nil {
		http.Error(w, "Gagal parsing data", http.StatusInternalServerError)
		return
	}
	if len(result) > 0 {
		kategori.ID = result[0].ID
	}

	w.Header().Set("Content-Type", "application/json")
	json.NewEncoder(w).Encode(models.Response{
		Success: true,
		Message: "Category created successfully",
		Data:    kategori,
	})
}
